tools/mason/MasonSearch.chpl

```diff
@@ -122,6 +119,25 @@ proc masonSearch(ref args: list(string)) {
   }
 }
 
+
+/* Sort the results in a order such that results that startWith needle
+   are displayed first */
+proc rankResults(results, query) {
+  use Sort;
+  record Comparator { }
+  proc Comparator.compare(a, b) {
+    if a.toLower().startsWith(query) && !b.toLower().startsWith(query) then return -1;
+    else if !a.toLower().startsWith(query) && b.toLower().startsWith(query) then return 1;
+    else return 1;
+  }
+  var cmp : Comparator;
+  var res = results.toArray();
+  sort(res, comparator = cmp);
+  if query != ".*" then
```

Comment:
This could also look like:

```chpl
// Sort alphabetically if query is empty
if query == ".*" then
  sort(res);
else
  sort(res, comparator=cmp);

return res;
```